emails/tests/views_tests.py

```diff
@@ -849,6 +781,151 @@ def test_invalid_message_id_is_forwarded(self, mock_logger: Mock) -> None:
         )
 
 
+class SNSNotificationRepliesTest(SNSNotificationTestBase):
+    """Tests for _sns_notification for replies from Relay users"""
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        # Create a premium user matching the s3_stored_replies sender
+        self.user = baker.make(User, email="source@sender.com")
+        self.user.profile.server_storage = True
+        self.user.profile.date_subscribed = datetime.now(tz=timezone.utc)
+        self.user.profile.last_engagement = datetime.now(tz=timezone.utc)
+        self.user.profile.save()
+        self.pre_reply_last_engagement = self.user.profile.last_engagement
+        upgrade_test_user_to_premium(self.user)
+
+        # Create a Reply record matching the s3_stored_replies headers
+        lookup_key, encryption_key = derive_reply_keys(
+            get_message_id_bytes("CA+J4FJFw0TXCr63y9dGcauvCGaZ7pXxspzOjEDhRpg5Zh4ziWg")
+        )
+        metadata = {
+            "message-id": str(uuid4()),
+            "from": "sender@external.example.com",
+        }
+        encrypted_metadata = encrypt_reply_metadata(encryption_key, metadata)
+        self.relay_address = baker.make(
+            RelayAddress, user=self.user, address="a1b2c3d4"
+        )
+        Reply.objects.create(
+            lookup=b64_lookup_key(lookup_key),
+            encrypted_metadata=encrypted_metadata,
+            relay_address=self.relay_address,
+        )
+
+        get_message_content_patcher = patch("emails.views.get_message_content_from_s3")
+        self.mock_get_content = get_message_content_patcher.start()
+        self.addCleanup(get_message_content_patcher.stop)
+
+    def success_test_implementation(self, text: str, expected_fixture_name: str) -> str:
+        """The headers of a reply refer to the Relay mask."""
+
+        self.mock_get_content.return_value = create_email_from_notification(
+            EMAIL_SNS_BODIES["s3_stored_replies"], text=text
+        )
+
+        # Successfully reply to a previous sender
+        response = _sns_notification(EMAIL_SNS_BODIES["s3_stored_replies"])
+        assert response.status_code == 200
+
+        self.mock_remove_message_from_s3.assert_called_once()
+        self.mock_get_content.assert_called_once()
+
+        sender, recipient, headers, email = self.get_details_from_mock_send_raw_email()
+        assert sender == "a1b2c3d4@test.com"
+        assert recipient == "sender@external.example.com"
+        assert headers == {
+            "Content-Type": headers["Content-Type"],
+            "Subject": "Re: Test Mozilla User New Domain Address",
+            "MIME-Version": "1.0",
+            "From": sender,
+            "Reply-To": sender,
+            "To": recipient,
+        }
```

Comment:
question (non-blocking): these assertion values are hard-coded in the test. should the test load these assertion values from the fixture rather than repeat them here?

Reply:
This isn't new code, just moved code. Replacing all the instances of `assert_email_equals` deserves it's own PR.